File: src/main/scala/org/cloudio/morpheus/mail/traditional/AttachmentValidator.scala
package org.cloudio.morpheus.mail.traditional

import java.util
import scala.collection.JavaConversions._

import org.cloudio.morpheus.mail.{Attachment, UserMail}

/**
 * Created by zslajchrt on 24/08/15.
 */
trait AttachmentValidator extends UserMail {

  abstract override def sendEmail(recipients: util.List[String], subject: String, message: String, attachments: util.List[Attachment]): Unit = {
    for (att <- attachments) {
      validateAttachment(att)
    }
    super.sendEmail(recipients, subject, message, attachments)
  }

  private def validateAttachment(att: Attachment): Unit = {
    // todo
  }
}

File: src/main/java/org/cloudio/morpheus/mail/AttachmentValidator.java
package org.cloudio.morpheus.mail;

import java.util.List;

/**
 * Created by zslajchrt on 24/08/15.
 */
public class AttachmentValidator implements UserMail {

    private final UserMail userMail;

    public AttachmentValidator(UserMail userMail) {
        this.userMail = userMail;
    }

    public void sendEmail(List<String> recipients, String subject, String message, List<Attachment> attachments) {

        for (Attachment attachment : attachments) {
            validateAttachment(attachment);
        }
        userMail.sendEmail(recipients, subject, message, attachments);
    }

    private void validateAttachment(Attachment attachment) {
        // todo
    }
}
